Corona-Warn-App/src/main/java/de/rki/coronawarnapp/statistics/local/source/LocalStatisticsParser.kt

```diff
@@ -0,0 +1,77 @@
+package de.rki.coronawarnapp.statistics.local.source
+
+import dagger.Reusable
+import de.rki.coronawarnapp.server.protocols.internal.stats.KeyFigureCardOuterClass
+import de.rki.coronawarnapp.server.protocols.internal.stats.LocalStatisticsOuterClass
+import de.rki.coronawarnapp.statistics.LocalIncidenceStats
+import de.rki.coronawarnapp.statistics.StatisticsData
+import org.joda.time.Instant
+import timber.log.Timber
+import javax.inject.Inject
+
+@Reusable
+class LocalStatisticsParser @Inject constructor() {
+
+    fun parse(rawData: ByteArray): StatisticsData {
+        val parsed = LocalStatisticsOuterClass.LocalStatistics.parseFrom(rawData)
+
+        val mappedFederalStates = parsed.federalStateDataList.mapNotNull { rawState ->
+            try {
+                val updatedAt = Instant.ofEpochSecond(rawState.updatedAt)
+                val stateIncidenceKeyFigure = rawState.sevenDayIncidence.toKeyFigure()
+
+                LocalIncidenceStats(
+                    updatedAt = updatedAt,
+                    keyFigures = listOf(stateIncidenceKeyFigure),
+                    federalState = rawState.federalState
+                ).also {
+                    Timber.tag(TAG).v("Parsed %s", it.toString().replace("\n", ", "))
+                    it.requireValidity()
+                }
+            } catch (e: Exception) {
+                Timber.tag(TAG).e("Failed to parse raw federal state: %s", rawState)
+                null
+            }
+        }
+
+        val mappedAdministrativeUnit = parsed.administrativeUnitDataList.mapNotNull { rawState ->
+            try {
+                val updatedAt = Instant.ofEpochSecond(rawState.updatedAt)
+                val administrativeUnitIncidenceKeyFigure = rawState.sevenDayIncidence.toKeyFigure()
+
+                val federalStateId = rawState.administrativeUnitShortId.toString().dropLast(3).toInt()
```

Comment:
Where does the `dropLast(3)` come from?

Reply:
https://github.com/corona-warn-app/cwa-app-tech-spec/pull/61/files#diff-b934511166d6db28e5cc528bcfc6df5ac63dd27eb1a9cc9192a61f2b16f4b73eR666